aggregator/list.go

```diff
@@ -196,15 +241,17 @@ func (l *metricList) Flush() {
 	l.RUnlock()
 
 	// Flush remaining bytes in the buffer.
-	if encoder := l.encoder.Encoder(); len(encoder.Bytes()) > 0 {
-		newEncoder := l.encoderPool.Get()
-		newEncoder.Reset()
-		l.encoder.Reset(newEncoder)
-		if err := l.flushHandler.Handle(NewRefCountedBuffer(encoder)); err != nil {
-			l.log.Errorf("flushing metrics error: %v", err)
-			l.metrics.flushErrors.Inc(1)
-		} else {
-			l.metrics.flushSuccess.Inc(1)
+	if flushType == consumeType {
+		if encoder := l.encoder.Encoder(); len(encoder.Bytes()) > 0 {
```

Comment:
nit: `if encoder := l.encoder.Encoder(); len(encoder.Bytes()) > 0 && flushType == consumeType {...}` to save an indentation?